scio-avro/src/main/scala/com/spotify/scio/avro/AvroIO.scala

```diff
@@ -166,46 +177,74 @@ final case class SpecificRecordIO[T <: SpecificRecord: ClassTag: Coder](path: St
     val cls = ScioUtil.classOf[T]
     val t = beam.AvroIO.write(cls)
     data.applyInternal(
-      avroOut(data, t, path, params.numShards, params.suffix, params.codec, params.metadata)
+      avroOut(data, t, paths.head, params.numShards, params.suffix, params.codec, params.metadata)
     )
     tap(())
   }
 
   override def tap(read: ReadP): Tap[T] =
-    SpecificRecordTap[T](ScioUtil.addPartSuffix(path))
+    SpecificRecordTap[T](ScioUtil.addPartSuffix(paths.head))
 }
 
-final case class GenericRecordIO(path: String, schema: Schema) extends AvroIO[GenericRecord] {
+object SpecificRecordIO {
+  def apply[T <: SpecificRecord: ClassTag: Coder](path: String): SpecificRecordIO[T] =
+    SpecificRecordIO(Iterable(path))
+}
+
+final case class GenericRecordIO(paths: Iterable[String], schema: Schema)
+    extends AvroIO[GenericRecord] {
   override type ReadP = Unit
   override type WriteP = AvroIO.WriteParam
 
-  override def testId: String = s"AvroIO($path)"
+  // TODO: support test with multiple paths
+  override def testId: String = s"AvroIO(${paths.head})"
 
   /**
    * Get an SCollection of [[org.apache.avro.generic.GenericRecord GenericRecord]] from an Avro
    * file.
    */
   override protected def read(sc: ScioContext, params: ReadP): SCollection[GenericRecord] = {
-    val t = beam.AvroIO
-      .readGenericRecords(schema)
-      .from(path)
-    sc.wrap(sc.applyInternal(t))
+    sc.wrap(
+      if (paths.size == 1) {
+        val t = beam.AvroIO
+          .readGenericRecords(schema)
+          .from(paths.head)
+        sc.applyInternal(t)
```

Comment:
I don't like that. Since multiple `paths` can only be used for reads, I think it should have its own read-only IO rather than shoehorning it into the existing IOs